pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -201,3 +250,55 @@ func extractFiles(body []byte, pluginName string, filePath string) error {
 func permissionsError(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "permission denied")
 }
+
+func isSymlink(file *zip.File) bool {
+	return file.Mode()&os.ModeSymlink == os.ModeSymlink
+}
+
+func extractSymlink(file *zip.File, filePath string) error {
+	// symlink target is the contents of the file
+	src, err := file.Open()
+	if err != nil {
+		return errutil.Wrap("Failed to extract file", err)
+	}
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, src)
+	if err != nil {
+		return errutil.Wrap("Failed to copy symlink contents", err)
+	}
+	err = os.Symlink(strings.TrimSpace(buf.String()), filePath)
+	if err != nil {
+		return errutil.Wrap(fmt.Sprintf("failed to make symbolic link for %v", filePath), err)
+	}
+	return nil
+}
+
+func extractFile(file *zip.File, filePath string) (err error) {
+	fileMode := file.Mode()
+	// This is entry point for backend plugins so we want to make them executable
+	if strings.HasSuffix(filePath, "_linux_amd64") || strings.HasSuffix(filePath, "_darwin_amd64") {
+		fileMode = os.FileMode(0755)
+	}
+
+	dst, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, fileMode)
```

Comment:
filePath is not checked in any way here. It wasn't checked before too, but before the zip files were generated from github sources.